app/react/src/server/framework-preset-react-docgen.ts

```diff
@@ -28,13 +29,23 @@ export function babel(config: TransformOptions, { typescriptOptions }: Storybook
   };
 }
 
+const tsconfigPath = './tsconfig.json';
+
 export function webpackFinal(config: Configuration, { typescriptOptions }: StorybookOptions) {
-  const { reactDocgen, reactDocgenTypescriptOptions } = typescriptOptions;
+  const { reactDocgen, reactDocgenTypescriptOptions = {} } = typescriptOptions;
```

Comment:
This is already the default, it's set in `defaults.js` :)
https://github.com/storybookjs/storybook/blob/next/lib/core/src/server/config/defaults.js

We added it very recently. However, if it's not working for you - let me know, as that would be a bug.